README.md

```diff
@@ -26,6 +28,69 @@ func main() {
 }
 ```
 
+Note: While the [source for this](tests/engine/testdata/fac.wat) is in the
+WebAssembly 1.0 (20191205) Text Format, it could have been written in another
+language that compiles to (targets) WebAssembly, such as AssemblyScript/C/C++/Rust/TinyGo/Zig.
+
+## Deeper dive
+
+The former example is a pure function. While a good start, you probably are
+wondering how to do something more realistic, like read a file. WebAssembly
+Modules (Wasm) are sandboxed similar to containers. They can't read anything
+on your machine unless you explicitly allow it.
+
+System access is defined by an emerging specification called WebAssembly
+System Interface ([WASI](https://github.com/WebAssembly/WASI)). WASI defines
+how WebAssembly programs interact with the host embedding them. For example,
+WASI defines functions for reading the time, or a random number. Let's dig
+deeper into this with file I/O.
+
+Since we are Go developers, let's use Go's `fs.FS` interface to connect the
+dots. This [example](examples/testdata/file_system.wasm) allows a WebAssembly Module access read an `embed.FS` you
+define:
+
+```golang
```

Comment:
I like this as an examples directory but it has too much going on for the top-level README. I'll think about an alternative while moving along!